src/main/java/uk/gov/ons/census/action/messaging/ActionFulfilmentReceiver.java

```diff
@@ -0,0 +1,100 @@
+      RabbitTemplate rabbitTemplate, CaseService caseService, CaseRepository caseRepository) {
+    this.rabbitTemplate = rabbitTemplate;
+    this.caseService = caseService;
+    this.caseRepository = caseRepository;
+  }
+
+  @Transactional
+  @ServiceActivator(inputChannel = "actionFulfilmentInputChannel")
+  public void receiveEvent(ResponseManagementEvent responseManagementEvent) {
+    UUID caseId =
+        UUID.fromString(responseManagementEvent.getPayload().getFulfilmentRequest().getCaseId());
+    Optional<Case> fulfilmentCase = caseRepository.findByCaseId(caseId);
+    log.with("caseId", caseId).debug("Fulfilment Requested Event");
+    if (fulfilmentCase.isEmpty()) {
+      log.with("caseId", caseId).error("Cannot find Case for fulfilment request.");
+      throw new RuntimeException(
+          String.format("Cannot find case %s for fulfilment request.", caseId));
+    }
+    String packCode =
+        responseManagementEvent.getPayload().getFulfilmentRequest().getFulfilmentCode();
+    Optional<Integer> questionnaireType = determineQuestionnaireType(packCode);
+    if (questionnaireType.isEmpty()) {
+      log.with("caseId", caseId)
+          .with("PackCode", packCode)
+          .error("Unknown packCode in fulfilment request.");
+      throw new RuntimeException(
+          String.format(
+              "Unknown packCode %s in fulfilment request for caseId %s", packCode, caseId));
+    }
+    UacQidDTO uacQid = caseService.getUacQid(questionnaireType.get().toString());
+    PrintFileDto printFile =
+        populatePrintFileDto(fulfilmentCase.get(), uacQid, responseManagementEvent);
+    rabbitTemplate.convertAndSend(outboundExchange, outboundPrinterRoutingKey, printFile);
+  }
+
+  private PrintFileDto populatePrintFileDto(
+      Case fulfilmentCase, UacQidDTO uacQid, ResponseManagementEvent event) {
+    PrintFileDto printFileDto = new PrintFileDto();
+    printFileDto.setAddressLine1(fulfilmentCase.getAddressLine1());
+    printFileDto.setAddressLine2(fulfilmentCase.getAddressLine2());
+    printFileDto.setAddressLine3(fulfilmentCase.getAddressLine3());
+    printFileDto.setTownName(fulfilmentCase.getTownName());
+    printFileDto.setPostcode(fulfilmentCase.getPostcode());
+    printFileDto.setQid(uacQid.getQid());
+    printFileDto.setUac(uacQid.getUac());
+    printFileDto.setTitle(event.getPayload().getFulfilmentRequest().getContact().getTitle());
+    printFileDto.setForename(event.getPayload().getFulfilmentRequest().getContact().getForename());
+    printFileDto.setSurname(event.getPayload().getFulfilmentRequest().getContact().getSurname());
+    printFileDto.setBatchId(UUID.randomUUID().toString());
+    printFileDto.setBatchQuantity(1);
+    printFileDto.setPackCode(event.getPayload().getFulfilmentRequest().getFulfilmentCode());
+    printFileDto.setActionType(ActionType.P_OR_HX.name());
+    return printFileDto;
+  }
+
+  private static Map<String, Integer> fulfilmentCodeToQuestionnaireType =
```

Comment:
might as well make this final too

Reply:
oh and move it up to the top of the class with the other member variables

Reply:
Done.